internal/batches/executor/run_steps.go

```diff
@@ -179,10 +181,10 @@ func runSteps(ctx context.Context, opts *executionOpts) (result execution.Result
 			return execResult, nil, errors.Wrap(err, "getting diff produced by step")
 		}
 		stepResult := execution.AfterStepResult{
-			StepIndex:          i,
-			Diff:               string(stepDiff),
-			Outputs:            make(map[string]interface{}),
-			PreviousStepResult: stepContext.PreviousStep,
+			StepIndex:  i,
+			Diff:       string(stepDiff),
+			Outputs:    make(map[string]interface{}),
+			StepResult: result,
```

Comment:
This was just weirdly wrong :D this meant that we got.. the second last result from the cache entry?

Reply:
Hmmmmm, I guess this is on me. This part of the code is really tricky - I hope we do have tests covering all the edge cases 😬